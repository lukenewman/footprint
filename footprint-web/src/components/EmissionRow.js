import React from 'react';
import { InputNumber, Row, Col, Typography } from 'antd';
import 'antd/dist/antd.css';

const { Title } = Typography;

/*
  EmissionRow consists of a title, InputNumber, and unit label

  Props:
    emissionType (EmissionType): the EmissionType represented by this row
    existingInput (Number | null): the previous value that the user may have entered already
    onInputChanged ((value, String) => ()): callback invoked each time the InputNumber value changes
*/
class EmissionRow extends React.Component {
  onInputChanged = (value) => {
    this.props.onInputChanged(value, this.props.emissionType.id);
  }

  render() {
    let emissionType = this.props.emissionType;
    return (
      <Row justify="center">
        <Col span={6}>
          <Title level={4}>{emissionType.name}</Title>
        </Col>
        <Col span={4}>
          <InputNumber defaultValue={this.props.existingInput} onChange={this.onInputChanged}></InputNumber>
        </Col>
        <Col span={4}>{emissionType.units}</Col>
      </Row>
    );
  }
}

export default EmissionRow;
