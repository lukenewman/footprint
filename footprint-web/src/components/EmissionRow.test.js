import React from "react";
import renderer from 'react-test-renderer';
import EmissionRow from "./EmissionRow.js";

test("<EmissionRow /> matches snapshot", () => {
  let emissionType = {
    name: "Natural Gas",
    units: "therms/year"
  };

  const component = renderer.create(
    <EmissionRow emissionType={emissionType} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});