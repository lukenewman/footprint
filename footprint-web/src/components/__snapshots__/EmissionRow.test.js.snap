// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmissionRow /> matches snapshot 1`] = `
<div
  className="ant-row ant-row-center"
  style={Object {}}
>
  <div
    className="ant-col ant-col-6"
    style={Object {}}
  >
    <h4
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Natural Gas
    </h4>
  </div>
  <div
    className="ant-col ant-col-4"
    style={Object {}}
  >
    <div
      className="ant-input-number"
      onBlur={[Function]}
      onCompositionEnd={[Function]}
      onCompositionStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
    >
      <div
        className="ant-input-number-handler-wrap"
      >
        <span
          aria-disabled={false}
          aria-label="Increase Value"
          className="ant-input-number-handler ant-input-number-handler-up"
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          role="button"
          unselectable="on"
        >
          <span
            aria-label="up"
            className="anticon anticon-up ant-input-number-handler-up-inner"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="up"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
              />
            </svg>
          </span>
        </span>
        <span
          aria-disabled={false}
          aria-label="Decrease Value"
          className="ant-input-number-handler ant-input-number-handler-down"
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          role="button"
          unselectable="on"
        >
          <span
            aria-label="down"
            className="anticon anticon-down ant-input-number-handler-down-inner"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        className="ant-input-number-input-wrap"
      >
        <input
          aria-valuenow={null}
          autoComplete="off"
          className="ant-input-number-input"
          onChange={[Function]}
          role="spinbutton"
          step={1}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="ant-col ant-col-4"
    style={Object {}}
  >
    therms/year
  </div>
</div>
`;
